okhttp/src/main/java/com/squareup/okhttp/ConnectionSpec.java

```diff
@@ -219,7 +219,7 @@ private ConnectionSpec supportedSpec(SSLSocket sslSocket) {
     private String[] tlsVersions;
     private boolean supportsTlsExtensions;
 
-    Builder(boolean tls) {
+    public Builder(boolean tls) {
```

Comment:
I'm quite reluctant to offer a naked ConnectionSpec. In the future we may add more stuff to ConnectionSpec: for example, four years ago we didn't have NPN, SNI, or SCSV. The defaults for these things will be different depending on whether you're going for maximum security (`MODERN_TLS`) or maximum compatibility (`COMPATIBLE_TLS`).

If we add an open ended default constructor, we're in trouble because we prevent the caller from choosing whether they prefer security or compatibility.

The current design is surprising because it makes you choose. That's deliberate!